src/commonMain/kotlin/exchange.dydx.abacus/state/v2/supervisor/SubaccountTransactionSupervisor.kt

```diff
@@ -147,6 +148,36 @@ internal class SubaccountTransactionSupervisor(
         return submitCancelOrder(orderId, marketId, callback, payload, analyticsPayload, uiClickTimeMs)
     }
 
+    internal fun cancelAllOrders(marketId: String?, callback: TransactionCallback): HumanReadableCancelAllOrdersPayload {
+        val payload = payloadProvider.cancelAllOrdersPayload(marketId)
+
+        payload.payloads.forEach { cancelPayload ->
+            val subaccount = stateMachine.state?.subaccount(subaccountNumber)
+            val existingOrder = subaccount?.orders?.firstOrNull { it.id == cancelPayload.orderId }
+                ?: throw ParsingException(
+                    ParsingErrorType.MissingRequiredData,
+                    "no existing order to be cancelled for $cancelPayload.orderId",
```

Comment:
hmm is it possible to hit this via race condition (i.e. if a user has two tabs open, clicks "cancel all orders" in one tab, and then manually cancels an existing order in another)? Instead of throwing an error, is it possible to just log it and gracefully continue?

Reply:
order should still exists even if it's canceled!